packages/application-generic/src/services/calculate-delay/calculate-delay.service.ts

```diff
@@ -35,27 +43,41 @@ export class CalculateDelayService {
       return delay;
     }
 
-    if (this.checkValidDelayOverride(overrides)) {
-      return this.toMilliseconds(
-        overrides.delay.amount as number,
-        overrides.delay.unit as DigestUnitEnum
-      );
-    }
+    if (isRegularDigest(stepMetadata.type)) {
+      if (this.checkValidDelayOverride(overrides)) {
+        return this.toMilliseconds(
+          overrides.delay.amount as number,
+          overrides.delay.unit as DigestUnitEnum
+        );
+      }
 
-    if (this.isRegularDigest(step.metadata.type)) {
-      const regularDigestMeta = step.metadata as IDigestRegularMetadata;
+      const regularDigestMeta = stepMetadata as IDigestRegularMetadata;
 
       return this.toMilliseconds(
         regularDigestMeta.amount,
         regularDigestMeta.unit
       );
     }
 
-    // TODO: timed digest
+    if (IS_ENTERPRISE && stepMetadata.type === DigestTypeEnum.TIMED) {
+      const timedDigestMeta = stepMetadata as IDigestTimedMetadata;
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      const { TimedDigestDelayService } = require('@novu/digest-schedule');
+
+      return TimedDigestDelayService.calculate({
+        unit: timedDigestMeta.unit,
+        amount: timedDigestMeta.amount,
+        atTime: timedDigestMeta.timed?.atTime,
+        weekDays: timedDigestMeta.timed?.weekDays,
+        monthDays: timedDigestMeta.timed?.monthDays,
+      });
+    }
+
     return 0;
   }
 
-  toMilliseconds(amount: number, unit: DigestUnitEnum): number {
+  private toMilliseconds(amount: number, unit: DigestUnitEnum): number {
```

Comment:
why do we make this private?

Reply:
because it's not used anywhere now only internally